src/hardware/serialport/serialport.cpp

```diff
@@ -1233,16 +1233,16 @@ bool CSerial::Putchar(uint8_t data, bool wait_dsr, bool wait_cts, uint32_t timeo
 	}
 	// wait for DSR+CTS on
 	if(wait_dsr||wait_cts) {
-		if(wait_dsr||wait_cts) {
+		if (wait_dsr && wait_cts) {
```

Comment:
This sure looks like a plain bug… I remember looking at it in the past but decided I have no idea if this is a correct solution (I couldn't find anything useful in SVN history - this code was part of large "code dump" commit).

But if a second programmer agrees this looks like a solution here, I am ok with merging this in :)

Also: this has 0 chance of causing regressions, as all places where `CSerial::Putchar` is called pass: wait_dsr = true, wair_cts = true.

Request: please update the commit message to indicate what was the problem with this code, as otherwise it might not be clear to people who haven't seen the static analysis issue:

```
Static analysis detected dead code in 'else' statements in here.
```
(or something similar)

Reply:
Took a second pass at it. Turned out to be much clearer to describe and implement.

In summary: if the line's been configured with wait-states, then we need to keep looping (and waiting) until the other end is done processing what we previously sent and eventually sets its ready-bits to true.